packages/keyring-controller/CHANGELOG.md

```diff
@@ -6,6 +6,48 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [6.0.0]
+### Added
+- Add `KeyringControllerMemState` type ([#1378](https://github.com/MetaMask/core/pull/1378))
+  - This is like the previous `KeyringMemState` but without `encryptionKey` or `encryptionSalt`
+- Add messenger events `KeyringController:lock` and `KeyringController:unlock`, emitted when the inner EthKeyringController is locked/unlocked ([#1378](https://github.com/MetaMask/core/pull/1378))
+  - Also add corresponding types `KeyringControllerLockEvent` and `KeyringControllerUnlockEvent`
+- Add `KeyringController:accountRemoved` event, fired whenever an account is removed through `removeAccount` ([#1416](https://github.com/MetaMask/core/pull/1416))
+
+### Changed
+- **BREAKING:** Update constructor to take a single argument, an options bag, instead of three arguments ([#1378](https://github.com/MetaMask/core/pull/1378))
+- **BREAKING:** Update controller so state is now accessible via `controller.state` instead of `controller.store.getState()` ([#1378](https://github.com/MetaMask/core/pull/1378))
+- **BREAKING:** Update KeyringController to take a required `messenger` option ([#1378](https://github.com/MetaMask/core/pull/1378))
+  - The messenger will now publish `KeyringController:stateChange` on state changes thanks to BaseControllerV2
+  - The messenger features a `KeyringController:getState` action thanks to BaseControllerV2
+  - Add types `KeyringControllerGetStateAction` and `KeyringControllerStateChangeEvent` for the above
+  - Add type `KeyringControllerMessenger`
+- **BREAKING:** Update `keyringState` property in the return value of several methods from a type of `KeyringMemState` to `KeyringControllerMemState` ([#1378](https://github.com/MetaMask/core/pull/1378))
+  - The affected methods are:
+    - `addNewAccount`
+    - `addNewAccountWithoutUpdate`
+    - `createNewVaultAndRestore`
+    - `createNewVaultAndKeychain`
+    - `importAccountWithStrategy`
+    - `removeAccount`
+    - `setLocked`
+    - `submitEncryptionKey`
+    - `submitPassword`
+  - The new type omits `vault`, `encryptionKey`, and `encryptionSalt`
+- **BREAKING:** Remove `KeyringState`, `KeyringMemState`, and `KeyringConfig` in favor of new types `KeyringControllerState`, `KeyringControllerActions`, `KeyringControllerEvents`, and `KeyringControllerOptions` ([#1378](https://github.com/MetaMask/core/pull/1378))
```

Comment:
The `KeyringMemState` type is still around; it's used as a return value for certain methods

```suggestion
- **BREAKING:** Remove `KeyringState` and `KeyringConfig` in favor of new types `KeyringControllerState`, `KeyringControllerActions`, `KeyringControllerEvents`, and `KeyringControllerOptions` ([#1378](https://github.com/MetaMask/core/pull/1378))
```

Reply:
Oh wait, nevermind, it was renamed.

Reply:
I guess we should mention the new name though:
```suggestion
- **BREAKING:** Remove `KeyringState`, `KeyringMemState`, and `KeyringConfig` in favor of new types `KeyringControllerState`, `KeyringControllerMemState`, `KeyringControllerActions`, `KeyringControllerEvents`, and `KeyringControllerOptions` ([#1378](https://github.com/MetaMask/core/pull/1378))
```

Reply:
You're right, I'm missing KeyringControllerMemState from this list. I had added it to the Added section but that's inconsistent with the other items here. Now that type is only contained in this entry: [0836f80](https://github.com/MetaMask/core/pull/1426/commits/0836f8020c460543ca87f986113f37019dc6ea42)